File: c++/383. Ransom Note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int letters1[26] = {0}, letters2[26] = {0};
        for (auto &ch : ransomNote)
            letters1[ch-'a']++;
        for (auto &ch : magazine)
            letters2[ch-'a']++;
        
        for (int i = 0; i < 26; i++) {
            if (letters2[i] < letters1[i])
                return false;
        }
        return true;
    }
};

File: go/src/leetcode/383. Ransom Note/383. Ransom Note.go
/**
@author ZhengHao Lou
Date    2021/12/04
*/
package main

/**
https://leetcode-cn.com/problems/ransom-note/
 */
const N = 26
func canConstruct(ransomNote string, magazine string) bool {

	var mapping func(string) [N]int
	mapping = func(s string) [N]int {
		m := [N]int{}
		for i := range s {
			m[int(s[i] - 'a')]++
		}
		return m
	}

	mr, ma := mapping(ransomNote), mapping(magazine)
	for i := range mr {
		if ma[i] < mr[i] {
			return false
		}
	}
	return true
}
